build.yaml

```diff
@@ -424,83 +474,114 @@ common_cli:
 #   * clean_repo: (enable if we need to clean old files owned by root)
 #
 
+# run_trial_modules contains a list of all the modules we want to execute
+# through trial.
+run_trial_modules: &run_trial_modules
+  - flocker.acceptance
+  - flocker.apiclient
+  - flocker.ca.functional
+  - flocker.ca.test
+  - flocker.cli
+  - flocker.common
+  - flocker.control
+  - flocker.dockerplugin
+  - flocker.node.agents
+  - flocker.node.test
+  # TODO:
+  # one of the functional tests is hanging, so we split the functional
+  # tests and comment out the subset that is hanging
+  # - flocker.node.functional
+  # - flocker.node.functional.test_docker
+  - flocker.node.functional.test_script
+  - flocker.node.functional.test_deploy
+  - flocker.provision
+  - flocker.restapi
+  - flocker.route
+  - flocker.test
+  - flocker.testtools
+  # flocker.volume needs to run as root due to ZFS calls
+  # - flocker.volume
+
+# run_trial_cli contains a list of all the CLI yaml anchors we want to
+# execute as part of our run_trial_tasks
+run_trial_cli: &run_trial_cli [
+  *hashbang,
+  *setup_pip_cache,
+  *cleanup,
+  *setup_venv,
+  *setup_flocker_modules,
+  *setup_coverage,
+  *setup_aws_env_vars,
+  *run_trial_with_coverage,
+  *run_coverage,
+  *convert_results_to_junit ]
+
+run_trial_cli_as_root: &run_trial_cli_as_root [
+  *hashbang,
+  *setup_pip_cache,
+  *cleanup,
+  *setup_venv,
+  *setup_flocker_modules,
+  *setup_coverage,
+  *setup_aws_env_vars,
+  *run_trial_with_coverage_as_root,
+  *run_coverage,
+  *convert_results_to_junit ]
```

Comment:
I wonder if it would be more sensible to just collect things from `*hashbang` to `*setup_coverage` and then separately `run_coverage` and `convert_results_to_junit`, since there appears to be at least a third variation at https://github.com/ClusterHQ/flocker/pull/1945/files#diff-5b123ecec7bf9d216a1323f790a0602aL513 that varies the middle bit.